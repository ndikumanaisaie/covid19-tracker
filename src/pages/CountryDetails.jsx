import React from 'react';
import { useParams, useNavigate } from 'react-router';
import { useSelector } from 'react-redux';
import { BsFillArrowLeftCircleFill } from 'react-icons/bs';
import Spinner from '../components/Spinner';

const CountryDetails = () => {
  const { covidData, status } = useSelector((state) => ({ ...state.covidData }));
  const { id } = useParams();

  const navigate = useNavigate();

  const countryData = covidData.find((data) => data.id === id);

  if (status === 'loading') {
    return <Spinner />;
  }

  return (
    <div className="card-details-container">
      <BsFillArrowLeftCircleFill className="back-btn" onClick={() => navigate('/')} />
      <div className="card-details">
        <p>Country</p>
        <p className="country">{countryData?.Country}</p>
      </div>
      <div className="card-details">
        <p>Total Deaths</p>
        <p>{countryData?.TotalDeaths}</p>
      </div>
      <div className="card-details">
        <p>Total Recovered</p>
        <p>{countryData?.TotalRecovered}</p>
      </div>
      <div className="card-details">
        <p>Active Cases</p>
        <p>{countryData?.ActiveCases}</p>
      </div>
      <div className="card-details">
        <p>Total Tests</p>
        <p>{countryData?.TotalTests}</p>
      </div>
      <div className="card-details">
        <p>Serious or Critical</p>
        <p>{countryData?.Serious_Critical}</p>
      </div>
      <div className="card-details">
        <p>Tests per 1M Population</p>
        <p>{countryData?.Tests_1M_Pop}</p>
      </div>
    </div>

  );
};

export default CountryDetails;
