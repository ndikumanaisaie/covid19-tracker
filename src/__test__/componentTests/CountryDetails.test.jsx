import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import CountryDetails from '../../pages/CountryDetails';
import store from '../../redux/configureStore';

describe('CountryDetails page', () => {
  it('it renders correctly', () => {
    const details = renderer
      .create(
        <Provider store={store}>
          <CountryDetails />
        </Provider>,
      )
      .toJSON();
    expect(details).toMatchSnapshot();
  });
});
