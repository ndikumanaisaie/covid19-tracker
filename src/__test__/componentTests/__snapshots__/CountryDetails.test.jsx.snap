// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetails page it renders correctly 1`] = `
<div
  className="card-details-container"
>
  <svg
    className="back-btn"
    fill="currentColor"
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
    />
  </svg>
  <div
    className="card-details"
  >
    <p>
      Country
    </p>
    <p
      className="country"
    />
  </div>
  <div
    className="card-details"
  >
    <p>
      Total Deaths
    </p>
    <p />
  </div>
  <div
    className="card-details"
  >
    <p>
      Total Recovered
    </p>
    <p />
  </div>
  <div
    className="card-details"
  >
    <p>
      Active Cases
    </p>
    <p />
  </div>
  <div
    className="card-details"
  >
    <p>
      Total Tests
    </p>
    <p />
  </div>
  <div
    className="card-details"
  >
    <p>
      Serious or Critical
    </p>
    <p />
  </div>
  <div
    className="card-details"
  >
    <p>
      Tests per 1M Population
    </p>
    <p />
  </div>
</div>
`;
